hooks/addon

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+set -eu
+vault="secret/$GENESIS_VAULT_PREFIX"
+alias="$GENESIS_ENVIRONMENT-blacksmith"
+
+ip=$(lookup params.ip)
+port=$(lookup params.port 3000)
+export SB_BROKER_URL=http://$ip:$port
+export SB_BROKER_USERNAME=blacksmith
+export SB_BROKER_PASSWORD=$(safe read $vault/broker:password)
+
+export BOSH_ENVIRONMENT=https://$(lookup params.ip):25555
+export BOSH_CA_CERT=$(safe read $vault/tls/ca:certificate)
+export BOSH_CLIENT="admin"
+export BOSH_CLIENT_SECRET="$(safe read ${vault}/users/admin:password)"
+
+list() {
+  echo "The following addons are defined:"
+  echo
+  echo "  visit     Open the Blacksmith Web Management Console"
+  echo "            in your browser (requires macOS)"
+  echo
+  echo "  register  Register this Blacksmith Broker with one of"
+  echo "            your Genesis-deployed Cloud Foundry instances"
+  echo
+  echo "            Interprets its first argument as the name of"
+  echo "            the Cloud Foundry environment.  If no argument"
+  echo "            is given, the current environment name is used"
+  echo
+  echo "  bosh      Set up a local alias for the Blacksmith"
+  echo "            BOSH director and log into it"
+  echo
+  echo "  eden      Interact with the broker via 'eden'"
+  echo "            (must be installed separately)"
+  echo
+  echo "  curl      Issue raw HTTP requests (via curl) against"
+  echo "            the Blacksmith Broker, with authentication"
+  echo
+}
+
+bosh() {
+  command $GENESIS_BOSH_COMMAND "$@"
+  return $?
+}
+
+has_alias() {
+  bosh envs | grep -q "^$BOSH_ENVIRONMENT\\t$alias\\t"
+  return $?
+}
+
+setup_alias() {
+  bosh alias-env --tty $alias | grep -v '^User'
+}
+
+is_logged_in() {
+  local user
+  user="$(bosh env --json | jq -Mr ".Tables[0].Rows[0].user")"
+  if [[ "$user" == "null" || "$user" == "(not logged in)" ]] ; then
+    return 1
+  fi
+  if [[ "$user" != "admin" ]] ; then
+    describe >&2 "Logged in as #C{$user}, expected to be #C{admin}"
+    return 2
+  fi
+  describe "Logged in as #C{$user}..."
+  return 0
+}
+
+login() {
+  bosh logout >/dev/null 2>&1
+
+  echo "Logging you in as user 'admin'..."
+  printf "%s\n%s\n" admin "$(safe read $vault/users/admin:password)" | \
+    BOSH_CLIENT="" BOSH_CLIENT_SECRET="" bosh login
+}
+
+case $GENESIS_ADDON_SCRIPT in
+list)
+  list
+  exit 0
+  ;;
+
+visit)
+  if ! command -v open >/dev/null 2>&1; then
+    echo "The 'visit' addon script only works on macOS, currently."
+    exit 1
+  fi
+  open "http://$SB_BROKER_USERNAME:$SB_BROKER_PASSWORD@$ip:$port"
+  ;;
+
+bosh)
```

Comment:
Worth considering for future-proofing:

name this login-to-bosh, or something similar, so that we can reserve the `bosh` name to provide an addon named `bosh` that allows you to run bosh commands against the blacksmith bosh directly without polluting the users .bosh/config with an alias that they shouldn't directly be accessing regularly.

Reply:
That's not something I want to make easy.  I'd rather pollute the bosh config to give operators pause -- do I really _need_ to muck about with the director?